config/src/config/metrics_config.rs

```diff
@@ -1,40 +0,0 @@
-// Copyright (c) The Diem Core Contributors
-// SPDX-License-Identifier: Apache-2.0
-
-use serde::{Deserialize, Serialize};
-use std::path::PathBuf;
-
-#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
-#[serde(default, deny_unknown_fields)]
```

Comment:
I understand why we have the `deny_unknown_fields` on these config types but when we remove things like this we might run into issues when doing node updates since if configs aren't updated we may end up with a node not being able to start up.

Reply:
I think the way to do this is to leave the config value present but unused.

Reply:
Happy to discuss, but we shouldn't leave around dead configurations. I see two options, rip the bandaid or improve documentation. I think one of the problems with ripping the bandaid is that the messaging is pretty awful since you have to carefully read the error message to see the invalid field. With that being said, what are your opinions?

Reply:
I went with an approach in the middle, delete the config and replace it with a struct with no fields and no serde strictness. That way if we deprecate other fields or configs, we can replace them and avoid keeping around checks on deprecated fields. I do not really agree with this since I would rather we crash and let user's ask questions or better understand the software but sentiment seems to be in favor of less friction :D.